http/transactions_middleware_test.go

```diff
@@ -0,0 +1,201 @@
+package http
+
+import (
+	"database/sql"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+
+	// nolint
+	. "github.com/onsi/ginkgo"
+	// nolint
+	. "github.com/onsi/gomega"
+
+	sqlmock "github.com/DATA-DOG/go-sqlmock"
```

Comment:
Instead of using a mock I'd prefer to run the tests using a real database. There is some infrastructure for that in the `testing` package. See the `leadership/flag_test.go` file for examples.